### PR TITLE
extract common env setting for inference launch scripts

### PR DESCRIPTION
**Pull request by d367wang**
This PR follows the comment https://github.com/opprop/checker-framework-inference/pull/290#discussion_r564872870, and extracts the common environmental variable setting for various downstream inference to avoid duplications in their launch scripts.

### CODE REVIEW

File: scripts/downstream-runtime-env-setup.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+#
```
Comment (xingweitian) [inline scripts/downstream-runtime-env-setup.sh:3]: What about remove this line?

File: scripts/downstream-runtime-env-setup.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+# This script setups the runtime environmental variables necessary to
+# run downstream type inference.
+#
+# Should be imported first by downstream inference starter script.
+# The following environmental variable should be specified in
+# downstream starter:
+# 1. checker & solver class
+# 2. classpath
+# 3. other inference arguments if necessary
+#
```
Comment (xingweitian) [inline scripts/downstream-runtime-env-setup.sh:13]: What about remove this line?

File: scripts/downstream-runtime-env-setup.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+ROOT=$(cd $(dirname "$0")/.. && pwd)
+CFI=$ROOT/checker-framework-inference
+AFU=$ROOT/annotation-tools/annotation-file-utilities
+Z3=$ROOT/z3/bin
+
+export PATH=$AFU/scripts:$Z3:$PATH
+
```
Comment (xingweitian) [inline scripts/downstream-runtime-env-setup.sh:21]: What about remove this line?

File: scripts/downstream-runtime-env-setup.sh [hunk 4]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+#
+# This script setups the runtime environmental variables necessary to
```
Comment (xingweitian) [inline scripts/downstream-runtime-env-setup.sh:4]: environmental -> environment

File: scripts/downstream-runtime-env-setup.sh [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+# run downstream type inference.
+#
+# Should be imported first by downstream inference starter script.
+# The following environmental variable should be specified in
```
Comment (xingweitian) [inline scripts/downstream-runtime-env-setup.sh:8]: environmental -> environment
variable -> variables

File: scripts/downstream-runtime-env-setup.sh [hunk 6]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+#
+# This script setups the runtime environmental variables necessary to
+# run downstream type inference.
+#
+# Should be imported first by downstream inference starter script.
```
Comment (xingweitian) [inline scripts/downstream-runtime-env-setup.sh:7]: `starter script` sounds weird...
What about `configuration scripts`?

File: scripts/downstream-runtime-env-setup.sh [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+# The following environmental variable should be specified in
+# downstream starter:
```
Comment (xingweitian) [inline scripts/downstream-runtime-env-setup.sh:9]: `downstream starter` sounds weird...
What about `downstream projects' configuration scripts`?

Comment (d367wang) [reply scripts/downstream-runtime-env-setup.sh:9]: I'm referring to the `run-XXX.sh` script here. What name do you think is good for this?

**Comment by xingweitian** [comment]
Should we merge this one?

**Pull request merged**